### ISSUE TITLE
Better representation of rounds and statuses for my studies?

### ISSUE THREAD

**Issue by kaj** [KTH/kpm#53]
What do we want to know about a students course in the front-end, and how do we want to organize that knowledge?  Currently (as of #52) we have:

```ts
export type APIStudies = {
  courses: Record<TCourseCode, TStudiesCourse>;
  programmes: Record<TProgrammeCode, TStudiesProgramme[]>;
};
export type TStudiesCourse = {
  course_code: TCourseCode;
  title: { sv: string; en: string };
  credits: number;
  creditUnitAbbr: string; // usually "hp", check other values!
  roles: TStuidesCourseInner[];
  rooms: TCanvasRoom[];
};
export type TStuidesCourseInner = {
  status?: "antagna" | "godkand" | "registrerade";
  year?: number;
  term?: "1" | "2";
  round?: string;
};
```

I think the roles attribute isn't great as is.  It will sometimes contain some entirely empty objects, and where the status is `"godkand"` the others will be empty.

So, one idea I had was to replace the `roles` attribute with:

```ts
export type TStudiesCourse = {
   // ... other fields ...
   admitted?: TCourseRound[];
   registered?: TCourseRound[];
   completed?: True;
}
export type TCourseRound =  {
   year: number;
   term: "1" | "2";
   round: string;
}
```

... but thinking some more, maybe we should instead unify the `roles` into:

```ts
export type TStudiesCourse = {
   // ... other fields ...
   rounds: TCourseRound[];
   completed?: True;
}
export type TCourseRound =  {
   status: "admitted" | "registered";
   year: number;
   term: "1" | "2";
   round?: string;
}
```

Where the `rounds` are "unified" so that each round is mentioned no more than once.  This could be also be typed like the following, which I think is a good reconstruction of the actual studadm model, but maybe the above is easier to use?

```ts
export type TStudiesCourse = {
   // ... other fields ...
   rounds: Record<TCourseRound, TStudentStatus>;
   completed?: True;
}
export type TCourseRound =  {
   year: number;
   term: "1" | "2";
   round?: string;
}
export type TStudentRoundStatus = "admitted" | "registered";
```

**Comment by jhsware** [comment]
This should be considered for https://github.com/KTH/kpm/pull/59

**Comment by jhsware** [comment]
After discussion we have decided to look at this separately from #59 because we need to update the UX.

**Comment by jhsware** [comment]
@kaj Consider adding proper:

- link to PM 
- link to registration
 
in my-studies-api for each course round.

**Comment by jhsware** [comment]
Fixed by #67
